elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -0,0 +1,267 @@
+        if (!DynamicConfigHelpers.isNullOrEmpty(sqlDefinition) && (sqlDefinition.contains(SEMI_COLON)
+                || containsDisallowedWords(sqlDefinition, SQL_SPLIT_REGEX, SQL_DISALLOWED_WORDS))) {
+            throw new IllegalStateException(
+                    "sql/definition provided in table config contain either ';' or one of these words: "
+                            + Arrays.toString(SQL_DISALLOWED_WORDS.toArray()));
+        }
+    }
+
+    /**
+     * Validate role name provided in security config.
+     * @param elideSecurityConfig ElideSecurityConfig
+     * @return boolean true if all role name passes validation else throw exception
+     */
+    private static boolean validateRoleInSecurityConfig(ElideSecurityConfig elideSecurityConfig) {
+        for (String role : elideSecurityConfig.getRoles()) {
+            if (containsDisallowedWords(role, ROLE_NAME_DISALLOWED_WORDS)) {
+                throw new IllegalStateException("ROLE provided in security config contain one of these words: "
+                        + Arrays.toString(ROLE_NAME_DISALLOWED_WORDS));
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Checks if input string has any of the disallowed words.
+     * @param String input string to validate
+     * @param keywords Array of disallowed words
+     * @return boolean true if input string does not contain any of the keywords
+     *         else false
+     */
+    private static boolean containsDisallowedWords(String str, String[] keywords) {
+        return Arrays.stream(keywords).anyMatch(str.toUpperCase(Locale.ENGLISH)::contains);
+    }
+
+    /**
+     * Checks if any word in the input string matches any of the disallowed words.
+     * @param String input string to validate
+     * @param splitter regex for splitting input string
+     * @param keywords Set of disallowed words
+     * @return boolean true if any word in the input string matches any of the
+     *         disallowed words else false
+     */
+    private static boolean containsDisallowedWords(String str, String splitter, Set<String> keywords) {
+        return DynamicConfigHelpers.isNullOrEmpty(str) ? false
+                : Arrays.stream(str.trim().toUpperCase(Locale.ENGLISH).split(splitter)).anyMatch(keywords::contains);
+    }
+
+    /**
+     * Define Arguments.
+     */
+    private static void defineArgs(Options options) {
+        Option configDirOption = new Option("c", "configDir", true,
+                "Path for Model Configs Directory.\n"
+                + "Expected Directory Structure:\n"
+                + "./security.hjson(optional)\n"
+                + "./variables.hjson(optional)\n"
+                + "./tables/\n"
+                + "./tables/table1.hjson\n"
+                + "./tables/table2.hjson\n"
+                + "./tables/tableN.hjson\n");
+        configDirOption.setRequired(true);
+        options.addOption(configDirOption);
```

Comment:
We should include another -h/--help for printing the usage

Reply:
added option for help. Had to make c/configDir optinal now and added explict check for that if args h/help is not passed